unit/util/interval_constraint.cpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************\
+
+Module: Unit tests for interval_constraint in
+        util/interval_constraint.cpp
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+#include <util/interval_constraint.h>
+#include <util/std_expr.h>
+#include <util/std_types.h>
+
+SCENARIO(
+  "interval_constraint with characters",
+  "[core][util][interval_constraint]")
+{
+  GIVEN("A string representing a character range and an character expression")
+  {
+    const integer_intervalt char_range('a', 'z');
+    const exprt expr = exprt("dummy", unsignedbv_typet(16));
```

Comment:
:pick: the style `const exprt expr{"dummy", unsignedbv_typet(16)};` is preferred :newspaper: